packages/signing/src/json-signature.ts

```diff
@@ -1,113 +1,142 @@
+     * Checks whether the given public key signed the given JSON transaction with its fields
+     * sorted alphabetically
+     * @param message JSON object
+     * @param signature Hex encoded signature (created with the Ethereum prefix)
+     * @param publicKey
+     */
+    export function isValidTransaction(message: JsonLike, chainId: string, signature: string, publicKey: string): boolean {
+        if (!publicKey) return true
+        else if (!signature) return false
+
+        const strMessage = normalizedJsonString(message)
+        const msgBytes = new TextEncoder().encode(strMessage)
+
+        return BytesSignature.isValidTransaction(msgBytes, chainId, signature, publicKey)
+    }
+
+    /**
+     * Returns the public key that signed the given JSON transaction, having its fields sorted alphabetically
      *
+     * @param message JSON object
      * @param signature Hex encoded signature (created with the Ethereum prefix)
-     * @param responseBody JSON object of the `response` or `error` fields
+     * @param expanded Whether the resulting public key should be expanded or not (default: no)
      */
-    export function recoverPublicKey(responseBody: any, signature: string, expanded: boolean = false): string {
+    export function recoverTransactionPublicKey(message: JsonLike, chainId: string, signature: string, expanded: boolean = false): string {
         if (!signature) throw new Error("Invalid signature")
-        else if (!responseBody) throw new Error("Invalid body")
+        else if (!message) throw new Error("Invalid body")
 
-        responseBody = JsonSignature.sort(responseBody)
-        const strBody = JSON.stringify(responseBody)
-        const bodyBytes = toUtf8Bytes(strBody)
-        const msgHash = hashMessage(bodyBytes)
-        const msgHashBytes = arrayify(msgHash)
+        const strMessage = normalizedJsonString(message)
+        const msgBytes = new TextEncoder().encode(strMessage)
 
-        const expandedPubKey = signingKeyRecoverPublicKey(msgHashBytes, signature)
-        if (expanded) return expandedPubKey
-        return computePublicKey(expandedPubKey, true)
+        return BytesSignature.recoverTransactionPublicKey(msgBytes, chainId, signature, expanded)
+    }
+
+    // HELPERS
+
+    export function normalizedJsonString(body: JsonLike): string {
+        const sortedRequest = sort(body)
+        return JSON.stringify(sortedRequest)
     }
 
     /**
-     * Returns a copy of the JSON data so that fields are ordered alphabetically and signatures are 100% reproduceable
-     * @param data Any valid JSON payload
-     */
-    export function sort(data: any) {
+    * Returns a copy of the JSON data so that fields are ordered alphabetically and signatures are 100% reproduceable
+    * @param data Any valid JSON payload
+    */
+    export function sort<T extends JsonLike>(data: T): T {
         switch (typeof data) {
             case "bigint":
```

Comment:
Just asking: Is intended to throw an Exception for `bigint`? We hadn't this behavior before.

Reply:
Bigint's are not a JSON standard serializable primitive, only numbers are supported.
So the alternative is to either serialize them as a numeric string, hex or base64